internal/terraform/node_resource_abstract_instance.go

```diff
@@ -926,13 +940,33 @@ func (n *NodeAbstractResourceInstance) plan(
 			ProposedNewState: proposedNewVal,
 			PriorPrivate:     priorPrivate,
 			ProviderMeta:     metaConfigVal,
+			DeferralAllowed:  ctx.Deferrals().DeferralAllowed(),
 		})
 	}
 	diags = diags.Append(resp.Diagnostics.InConfigBody(config.Config, n.Addr.String()))
 	if diags.HasErrors() {
 		return nil, nil, keyData, diags
 	}
 
+	if resp.Deferred != nil {
+		reqRep, reqRepDiags := getRequiredReplaces(priorVal, proposedNewVal, resp.RequiresReplace, n.ResolvedProvider.Provider, n.Addr)
+		diags = diags.Append(reqRepDiags)
+		if diags.HasErrors() {
+			return nil, nil, keyData, diags
+		}
+
+		action, _ := getAction(n.Addr, priorVal, resp.PlannedState, createBeforeDestroy, forceReplace, reqRep)
+		ctx.Deferrals().ReportResourceInstanceDeferred(n.Addr, resp.Deferred.Reason, &plans.ResourceInstanceChange{
+			Addr: n.Addr,
+			Change: plans.Change{
+				Action: action,
+				Before: unmarkedPriorVal,
+				After:  unmarkedConfigVal,
```

Comment:
We shouldn't used the unmarked values here. This will / might expose sensitive values elsewhere. You should reapply the marks before saving the change anywhere.